cluster/v1alpha1/helpers.go

```diff
@@ -55,24 +54,30 @@ type ClusterRolloutStatus struct {
 
 // RolloutResult contains the clusters to be rolled out and the clusters that have timed out.
 type RolloutResult struct {
-	// ClustersToRollout is a map where the key is the cluster name and the value is the ClusterRolloutStatus.
-	ClustersToRollout map[string]ClusterRolloutStatus
-	// ClustersTimeOut is a map where the key is the cluster name and the value is the ClusterRolloutStatus.
-	ClustersTimeOut map[string]ClusterRolloutStatus
+	// ClustersToRollout is a slice of ClusterRolloutStatus that will be rolled out.
+	ClustersToRollout []ClusterRolloutStatus
+	// ClustersTimeOut is a slice of ClusterRolloutStatus that are timeout.
+	ClustersTimeOut []ClusterRolloutStatus
+	// ClustersRemoved is a slice of ClusterRolloutStatus that are removed.
+	ClustersRemoved []ClusterRolloutStatus
```

Comment:
I'm assuming this is an array now so that the mapping to each cluster's name is clearer to the end user? I kind of liked the `map[string]ClusterRolloutStatus` for uniqueness and simplicity.